DeferredTests/DeferredTests.swift

```diff
@@ -305,4 +320,63 @@ class DeferredTests: XCTestCase {
         waitForExpectationsWithTimeout(0.5, handler: nil)
     }
     
+    func testIsFilledCanBeCalledMultipleTimesNotFilled() {
+        let d = Deferred<Int>()
+        XCTAssertFalse(d.isFilled)
+        XCTAssertFalse(d.isFilled)
+        XCTAssertFalse(d.isFilled)
+    }
+
+    func testIsFilledCanBeCalledMultipleTimesWhenFilled() {
+        let d = Deferred<Int>(value: 42)
+        XCTAssertTrue(d.isFilled)
+        XCTAssertTrue(d.isFilled)
+        XCTAssertTrue(d.isFilled)
+    }
+
+    // The QoS APIs do not behave as expected on the iOS Simulator, so we only
+    // run these tests on real devices. This check isn't the most future-proof;
+    // if there's ever another archiecture that runs the simulator, this will
+    // need to be modified.
+    //
+    // TODO: Add a clause for a tvOS test target
+    #if os(OSX) || (os(iOS) && !(arch(i386) || arch(x86_64)))
```

Comment:
I thought there must be a better way to handle this, and you probably tried, but I'm seeing mysterious behavior so I'll comment here anyway.

Things like `TARGET_OS_SIMULATOR` as defined in `TargetConditionals.h` are available to use in Swift, but from my testing:

1. using it with compile-time `#if` does not pass in the iPhone Simulator
1. using it with regular `if` at runtime *does* pass in the Simulator

What's up with that?